src/lib/db/transaction.ts

```diff
@@ -81,6 +86,25 @@ export function withTransactional<S>(
     return service;
 }
 
+export function withRollback<S>(
+    serviceFactory: (db: Knex) => S,
+    db: Knex,
+): WithRollback<S> {
+    const service = serviceFactory(db) as WithRollback<S>;
+
+    service.rollback = async <R>(fn: (service: S) => R) => {
+        const trx = await db.transaction({ isolationLevel: 'serializable' });
+        try {
+            const transactionService = serviceFactory(trx);
+            return fn(transactionService);
+        } finally {
+            await trx.rollback();
+        }
+    };
```

Comment:
Thanks for the feedback. I see your point and I will consider renaming it to sth else. I couldn't find a good name.

Reply:
Maybe even `withTemporaryTransaction` or something? 🤷🏼

Reply:
I have an issue with temporary/transient because all transactions are temporary for the duration of the transaction. I need something that explains that this transaction with be rolled back. `withRolledBackTransaction`?